### ISSUE TITLE
[BUG]: Build errors when declarations is true in tsconfig

### ISSUE THREAD

**Issue by ari-party** [roblox-ts/roblox-ts#2753]
![image](https://github.com/user-attachments/assets/9ac53fca-7815-48dc-9ab5-563b2ff5e488)

https://github.com/roblox-ts/roblox-ts/blob/576ad988c74ee1888109ded5f261d9b0ccd3aa7f/src/Project/transformers/builtin/transformPaths.ts#L271

```json
// tsconfig where error occurs
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "downlevelIteration": true,
    "jsx": "react",
    "jsxFactory": "React.createElement",
    "jsxFragmentFactory": "React.createFragment",
    "module": "commonjs",
    "moduleResolution": "Node",
    "noLib": true,
    "resolveJsonModule": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "moduleDetection": "force",
    "strict": true,
    "target": "ESNext",
    "typeRoots": ["node_modules/@rbxts"],

    "declaration": true,
    "rootDir": "src",
    "outDir": "dist",
    "baseUrl": "src",
    "tsBuildInfoFile": "dist/tsconfig.tsbuildinfo",
    "incremental": true
  }
}
```

Occurs on version `2.3.0-dev-576ad98`

**Comment by ari-party** [comment]
For those also experiencing this error, you can use [tsup](https://github.com/egoist/tsup) with the [`--dts-only`](https://tsup.egoist.dev/#emit-declaration-file-only) flag.

**Comment by Dionysusnu** [comment]
Can reproduce: seems to happen on all `export * from "<>"` statements, where `<>` is a relative path, or resolves using baseUrl (but not when it's a package specifier like `"@rbxts/baseplate"`)

**Comment by Dionysusnu** [comment]
The referenced method was removed as part of the TS 5.5 removal of deprecated import elision options - more specifically in https://github.com/microsoft/TypeScript/commit/7d504551ae670cd9a2358c9b6add40f92cb5b5f0

**Comment by ari-party** [comment]
Repository this occurs on, for other testers/contributors/members: https://github.com/grand-hawk/ui-components/blob/a178ead04d3d611f890d50061a1b36dd08aa8fb1/src/index.ts

**Comment by osyrisrblx** [comment]
Fixed by e6a6b3b
